### ISSUE TITLE
problem with gravity entries with trailing period

### ISSUE THREAD

**Issue by jpgpi250** [pi-hole/FTL#1330]
### Versions
  Pi-hole version is v5.9.1 (Latest: v5.9.1)
  AdminLTE version is v5.11 (Latest: v5.10.1)
  FTL version is v5.14 (Latest: v5.14)

### Platform
uname -a: Linux raspberrypi 5.15.32-v7+ #1538 SMP Thu Mar 31 19:38:48 BST 2022 armv7l GNU/Linux
Raspberry Pi OS Lite, Release date: April 4th 2022

### Expected behavior

I have a number of entries in the gravity database (2213) with a trailing period, example:
**mail.forblown.pl.**

I would expect pihole to block both **mail.forblown.pl.** (trailing period) and **mail.forblown.pl** (no trailing period)

pihole -q reports:
without trailing period: pihole -q mail.forblown.pl
 Match found in https://dbl.oisd.nl:
   mail.forblown.pl.
with trailng period: pihole -q mail.forblown.pl.
 Match found in https://dbl.oisd.nl:
   mail.forblown.pl.

this list changes frequently, so the entry may no longer be there, however, again, I have 2213 entries with a trailing period in the gravity table.

I would expect dig +short **mail.forblown.pl.** and dig +short **mail.forblown.pl** to report 0.0.0.0

### Actual behavior / bug

dig +short  **mail.forblown.pl.** and dig +short  **mail.forblown.pl** report 31.31.72.43 (the actual address)

### Steps to reproduce

1. ensure you have the entry **mail.forblown.pl.** (with trailing period) in the gravity database
2. run pihole -q on both **mail.forblown.pl** and **mail.forblown.pl.**
3. run dig on both **mail.forblown.pl** and **mail.forblown.pl.**

## Debug Token

debug token is: https://tricorder.pi-hole.net/FuKy6tcC/

### Additional context

removing the trailing period from the entry in the gravity database (and restarting pihole-FTL) fixes the problem.
NOT sure what is the most efficient solution:
- modify gravity.sh to always remove trailing periods.
- modify pihole-FTL.

**Comment by DL6ER** [comment]
FTL uses a balanced tree with string hashes to be able to scan the entire gravity table in sub-milliseconds even when there are million entries in it. The upside (speed) comes with an unavoidable drawback: only *exact* comparisons are possible. As we are utilizing (well-tested and well-optimized) SQLIte3 internals for this job, I wouldn't want to modify them in the SQLite3 binary, causing possible unintended side-effects somewhere else.

TL;DR: You suggestion

> modify gravity.sh to always remove trailing periods

will be the way to go here. I'll move this issue into the `pi-hole/pi-hole` repo.
